### PR TITLE
Starter none msg issue

### PR DESCRIPTION
**Pull request by Sreejith-Websphere**
#### What was fixed?  (Issue # or description of fix)

For https://github.com/openliberty/start.openliberty.io/issues/107

#### Were the changes tested on
- [ ] Firefox (Desktop)
- [ ] Safari (Desktop)
- [ ] Chrome (Desktop)
- [ ] Internet Explorer (Desktop)
- [ ] iOS (Mobile)
- [ ] Android (Mobile)
#### Running validation tools
- [ ] https://validator.w3.org/checklink
- [ ] https://validator.w3.org
- [ ] IBM Equal Access Accessibility Checker (https://www.ibm.com/able/toolkit/verify/automated)
- [ ] Lighthouse (in Chrome dev tools)

### CODE REVIEW

**Comment by kinueng** [comment]
Let's make sure we write Cypress test cases to ensure the message is shown when it makes sense

**Comment by Sreejith-Websphere** [comment]
@kinueng @aknguyen7 cypress test case added

File: src/test/cypressjs/cypress/integration/testOLStarterJakartaMPNone.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        cy.get('#starter_warnings p').should('exist') //message should be shown
+
+        cy.get('#Starter_MicroProfile_Version').select("None");
+        cy.get('#starter_warnings p').should('not.exist') //message should not be shown
+
+        cy.get('#Starter_MicroProfile_Version').select("1.4");
+        cy.get('#starter_warnings p').should('exist') //message should be shown
+
+        cy.get('#Starter_Jakarta_Version').select("None");
+        cy.get('#starter_warnings p').should('not.exist') //message should not be shown
```
Comment (aknguyen7) [inline src/test/cypressjs/cypress/integration/testOLStarterJakartaMPNone.js:74]: I assume this `#starter_warnings p` works? I'd normally see the syntax for cypress like `#starter_warings>p` instead

**Review by aknguyen7 — COMMENTED** [review]
the rest LGTM

File: src/test/cypressjs/cypress/integration/testOLStarterJakartaMPNone.js (hunk 1 above)
Comment (Sreejith-Websphere) [reply src/test/cypressjs/cypress/integration/testOLStarterJakartaMPNone.js:74]: Hi @aknguyen7 this cypress syntax works #starter_warnings p, below is the example I just changed the test case to fail, can see that syntax works

Screenshot 2022-07-11 at 1 23 43 PM

File: src/main/content/_assets/js/builds.js [hunk 2]
```diff
@@ -710,12 +711,15 @@ function validate_starter_inputs(event) {
                 if(EEVersionValue !== "None") {
                     curr_selected_mp_version = valuetoSelect = dependencies[d][dependencies[d].length - 1];
                 }
-                else{
-                    curr_selected_mp_version = valuetoSelect = starter_info[d].default;
-                }
                 var prev_selected_mp_version = options
                 .filter(':selected')
                 .text();
+                if ((EEVersionValue == "None")&&(prev_selected_mp_version == "None")) {
```
Comment (kinueng) [inline src/main/content/_assets/js/builds.js]:
```suggestion
                if((EEVersionValue === "None") && (prev_selected_mp_version === "None")) {
```

I would suggest always prefer `===` over `==` unless type conversion is needed. In this case, I believe the code is always dealing with String type and no type conversion is expected.

File: src/main/content/_assets/js/builds.js [hunk 3]
```diff
@@ -798,7 +788,15 @@ function validate_starter_inputs(event) {
             .text();
             for(var i=0; i<options.length; i++) {
                 if(options[i].value === EEVersion) {
-                    $(options[i]).prop('selected', true);
+                    if(mpVersionValue !== "None"){
```
Comment (kinueng) [inline src/main/content/_assets/js/builds.js]:
```suggestion
                    if(mpVersionValue !== "None") {
```

For code consistency

File: src/main/content/_assets/js/builds.js [hunk 4]
```diff
@@ -798,7 +788,15 @@ function validate_starter_inputs(event) {
...
+                        $(options[i]).prop('selected', true);
+                    }
+                    if((mpVersionValue == "None")&&(prev_selected_ee_version == "None")) {
```
Comment (kinueng) [inline src/main/content/_assets/js/builds.js]:
```suggestion
                    if((mpVersionValue === "None") && (prev_selected_ee_version === "None")) {
```

**Review by kinueng — CHANGES_REQUESTED** [review]

File: src/main/content/_assets/js/builds.js (hunk 2 above)
Comment (Sreejith-Websphere) [reply src/main/content/_assets/js/builds.js]: Done

**Review by kinueng — APPROVED** [review]
LGTM

**Pull request merged**